### PR TITLE
[WIP]  S3 select Json

### PR DESCRIPTION
**Pull request by sinhaashish**
[WIP]SQL select query for JSON is implemented.

## Description

Select * has been implemented , with partial feature of where clause. Other functionalities are yet to be added.

## Motivation and Context

It gives the SQL select feature for JSON documents.

## How Has This Been Tested?

Tested using AWS java client.

Tested Query
select * from S3Object AS aliasname 
select * from S3Object AS aliasname where aliasname.Name LIKE 'A%'

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
```
Error running mint automation
+ mkdir -p 6406-9999014/gopath/src/github.com/minio
+ git clone --quiet https://github.com/minio/minio.git 6406-9999014/gopath/src/github.com/minio/minio
++ cd 6406-9999014/gopath/src/github.com/minio/minio
++ git remote add minio https://github.com/minio/minio.git
++ cd 6406-9999014/gopath/src/github.com/minio/minio
++ git fetch --quiet minio pull/6406/head:pr6406
++ cd 6406-9999014/gopath/src/github.com/minio/minio
++ git checkout --quiet pr6406
++ mkdir -p certs
++ cd certs
++ ./cert -host localhost
2018/09/02 14:40:48 wrote public.crt
2018/09/02 14:40:48 wrote private.key
+ GOPATH=/home/mint-auto/mint-auto/6406-9999014/gopath
+ make -C 6406-9999014/gopath/src/github.com/minio/minio --quiet
pkg/s3select/input.go:30:2: cannot find package "github.com/json-iterator/go" in any of:
	/home/mint-auto/mint-auto/6406-9999014/gopath/src/github.com/minio/minio/vendor/github.com/json-iterator/go (vendor tree)
	/home/mint-auto/.gimme/versions/go1.10.2.linux.amd64/src/github.com/json-iterator/go (from $GOROOT)
	/home/mint-auto/mint-auto/6406-9999014/gopath/src/github.com/json-iterator/go (from $GOPATH)
make: *** [build] Error 1

```

**Comment by kannappanr** [comment]
@sinhaashish You have missed to vendorize `json` related packages

**Comment by minio-ops** [comment]
```
Error running mint automation
+ mkdir -p 6406-66f2e53/gopath/src/github.com/minio
+ git clone --quiet https://github.com/minio/minio.git 6406-66f2e53/gopath/src/github.com/minio/minio
++ cd 6406-66f2e53/gopath/src/github.com/minio/minio
++ git remote add minio https://github.com/minio/minio.git
++ cd 6406-66f2e53/gopath/src/github.com/minio/minio
++ git fetch --quiet minio pull/6406/head:pr6406
++ cd 6406-66f2e53/gopath/src/github.com/minio/minio
++ git checkout --quiet pr6406
++ mkdir -p certs
++ cd certs
++ ./cert -host localhost
2018/09/02 23:09:45 wrote public.crt
2018/09/02 23:09:45 wrote private.key
+ GOPATH=/home/mint-auto/mint-auto/6406-66f2e53/gopath
+ make -C 6406-66f2e53/gopath/src/github.com/minio/minio --quiet
vendor/github.com/json-iterator/go/config.go:10:2: cannot find package "github.com/modern-go/concurrent" in any of:
	/home/mint-auto/mint-auto/6406-66f2e53/gopath/src/github.com/minio/minio/vendor/github.com/modern-go/concurrent (vendor tree)
	/home/mint-auto/.gimme/versions/go1.10.2.linux.amd64/src/github.com/modern-go/concurrent (from $GOROOT)
	/home/mint-auto/mint-auto/6406-66f2e53/gopath/src/github.com/modern-go/concurrent (from $GOPATH)
vendor/github.com/json-iterator/go/any.go:6:2: cannot find package "github.com/modern-go/reflect2" in any of:
	/home/mint-auto/mint-auto/6406-66f2e53/gopath/src/github.com/minio/minio/vendor/github.com/modern-go/reflect2 (vendor tree)
	/home/mint-auto/.gimme/versions/go1.10.2.linux.amd64/src/github.com/modern-go/reflect2 (from $GOROOT)
	/home/mint-auto/mint-auto/6406-66f2e53/gopath/src/github.com/modern-go/reflect2 (from $GOPATH)
make: *** [build] Error 1

```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-tls.sh | :x: [more...](#a9c742e1-cd39-4949-b4d5-30bdf2703469) |
<a name="a9c742e1-cd39-4949-b4d5-30bdf2703469"></a>
### 6406-d6a6688/mint-tls.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         1
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8243fb49a55a:/mint/log /tmp/mint-logs'
(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-p1o63qtzi4jf5s",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 368,
  "error": "RequestError: send request failed\ncaused by: Put https://localhost/aws-sdk-go-test-p1o63qtzi4jf5s: dial tcp 127.0.0.1:443: getsockopt: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-tls.sh | :x: [more...](#b2b28da6-284b-4f93-ae9d-0666a587adc6) |
<a name="b2b28da6-284b-4f93-ae9d-0666a587adc6"></a>
### 6406-f644682/mint-tls.sh.log:
```
Running with
SERVER_ENDPOINT:      localhost
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         1
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9f418c0222e5:/mint/log /tmp/mint-logs'
(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-9t1kd9atomyzk3",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 340,
  "error": "RequestError: send request failed\ncaused by: Put https://localhost/aws-sdk-go-test-9t1kd9atomyzk3: dial tcp 127.0.0.1:443: getsockopt: connection refused",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```

**Review by nitisht — CHANGES_REQUESTED** [review]
@sinhaashish please check travis failure. Also pls remove the `debug` and `log.log` files - looks like they got committed inadvertently

**Comment by donatello** [comment]
There are linting errors in the travis build:

```
Running lint
/home/travis/gopath/src/github.com/minio/minio/pkg/s3select/input.go:78:1: comment on exported type JSONInput should be of the form "JSONInput ..." (with optional leading article)
/home/travis/gopath/src/github.com/minio/minio/pkg/s3select/input.go:88:6: exported type SelectQuery should have comment or be unexported
/home/travis/gopath/src/github.com/minio/minio/pkg/s3select/input.go:222:1: comment on exported function NewJsonInput should be of the form "NewJsonInput ..."
/home/travis/gopath/src/github.com/minio/minio/pkg/s3select/input.go:226:6: func NewJsonInput should be NewJSONInput
/home/travis/gopath/src/github.com/minio/minio/pkg/s3select/select.go:161:1: comment on exported method JSONInput.ParseSelect should be of the form "ParseSelect ..."
/home/travis/gopath/src/github.com/minio/minio/pkg/s3select/select.go:417:3: var readerJson should be readerJSON
Found 6 lint suggestions; failing.
make: *** [lint] Error 1
```

Please fix these.

File: pkg/s3select/funcEval.go [hunk 1]
```diff
@@ -186,6 +186,41 @@ func (reader *Input) evaluateFuncErr(myVal *sqlparser.FuncExpr) error {
 	return nil
 }
 
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% //
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%     PARSE SELECT FOR JSON STARTS  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% //
+//%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% //
+// evaluateFuncErr is a function that flags errors in nested functions.
```
Comment (donatello) [inline pkg/s3select/funcEval.go]: Minio codebase doesn't use this kind of section marking comment style - please change these so they look like the rest of the code.

File: pkg/s3select/helpers.go [hunk 2]
```diff
@@ -675,6 +748,30 @@ func (reader *Input) colNameErrs(columnNames []string) error {
...
+		// 	tempInt, _ := strconv.Atoi(columnNames[i])
+		// 	if tempInt > len(reader.Header()) || tempInt == 0 {
+		// 		return ErrInvalidColumnIndex
+		// 	}
+		// } else {
+		// 	if reader.options.HeaderOpt && !stringInSlice(columnNames[i], reader.Header()) {
+		// 		return ErrMissingHeaders
+		// 	}
+		// }
+	}
```
Comment (donatello) [inline pkg/s3select/helpers.go:762]: This function is incomplete? (Please mark PR as WIP)

File: pkg/s3select/jsonHelpers.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
+/*
```
Comment (donatello) [inline pkg/s3select/jsonHelpers.go]: This file name should be `json-helpers.go` - we don't use camelcase filenames.

**Review by donatello — CHANGES_REQUESTED** [review]

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#47f52178-0a7c-48af-9cd1-c158ffbfee9e) |
<a name="47f52178-0a7c-48af-9cd1-c158ffbfee9e"></a>
### 6406-b7df140/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32321
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 801eba15b1e7:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 2 minutes and 42 seconds
(2/13) Running aws-sdk-go tests ... done in 1 seconds
(3/13) Running aws-sdk-java tests ... done in 2 seconds
(4/13) Running aws-sdk-php tests ... done in 42 seconds
(5/13) Running aws-sdk-ruby tests ... done in 39 seconds
(6/13) Running mc tests ... done in 7 minutes and 33 seconds
(7/13) Running minio-dotnet tests ... FAILED in 5 minutes and 4 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.AreEqual failed. Expected:<0x8D61979D5DC5382-1>. Actual:<0x8D61979D56978B5-1>. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.AreEqual failed. Expected:<0x8D61979D5DC5382-1>. Actual:<0x8D61979D56978B5-1>. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual[T](T expected, T actual, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.CopyObject_Test3(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 960
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 223

Executed 6 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#cc851c36-be0f-4c2b-ad87-6ef6224cd194) |
| mint-dist-xl.sh | :x: [more...](#9defc70d-1cbb-4f64-89bf-74a6a0191411) |
<a name="9defc70d-1cbb-4f64-89bf-74a6a0191411"></a>
### 6406-a344504/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31369
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp fd7bf14c1d59:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 59 seconds
(2/13) Running aws-sdk-go tests ... done in 0 seconds
(3/13) Running aws-sdk-java tests ... done in 4 seconds
(4/13) Running aws-sdk-php tests ... done in 43 seconds
(5/13) Running aws-sdk-ruby tests ... done in 7 seconds
(6/13) Running mc tests ... done in 29 seconds
(7/13) Running minio-dotnet tests ... done in 2 minutes and 34 seconds
(8/13) Running minio-go tests ... done in 1 minutes and 12 seconds
(9/13) Running minio-java tests ... done in 1 minutes and 24 seconds
(10/13) Running minio-js tests ... done in 37 seconds
(11/13) Running minio-py tests ... FAILED in 30 minutes and 6 seconds
{
  "args": {
    "bucket_name": "minio-py-test-126d732c-9113-49ff-92c4-c1a96b13cf90",
    "location": "default value ('us-east-1')"
  },
  "name": "minio-py:test_make_bucket_default_region",
  "function": "make_bucket(bucket_name, location)",
  "status": "FAIL",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 600, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 384, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 2, in raise_from\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 380, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.5/http/client.py\", line 1197, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.5/http/client.py\", line 297, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.5/http/client.py\", line 266, in _read_status\n    raise RemoteDisconnected(\"Remote end closed connection without\"\nhttp.client.RemoteDisconnected: Remote end closed connection without response\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 167, in test_make_bucket_default_region\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.5/dist-packages/minio/api.py\", line 288, in make_bucket\n    headers=headers)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/poolmanager.py\", line 322, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 638, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/util/retry.py\", line 398, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minikube', port=31369): Max retries exceeded with url: /minio-py-test-126d732c-9113-49ff-92c4-c1a96b13cf90/ (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1720, in main\n    test_make_bucket_default_region(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 175, in test_make_bucket_default_region\n    raise Exception(err)\nException: HTTPConnectionPool(host='minikube', port=31369): Max retries exceeded with url: /minio-py-test-126d732c-9113-49ff-92c4-c1a96b13cf90/ (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n",
  "message": "HTTPConnectionPool(host='minikube', port=31369): Max retries exceeded with url: /minio-py-test-126d732c-9113-49ff-92c4-c1a96b13cf90/ (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))",
  "duration": 1806016
}

Executed 10 out of 13 tests successfully.
```
<a name="cc851c36-be0f-4c2b-ad87-6ef6224cd194"></a>
### 6406-a344504/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32075
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 732753098bbc:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 2 minutes and 46 seconds
(2/13) Running aws-sdk-go tests ... done in 2 seconds
(3/13) Running aws-sdk-java tests ... done in 4 seconds
(4/13) Running aws-sdk-php tests ... done in 45 seconds
(5/13) Running aws-sdk-ruby tests ... done in 39 seconds
(6/13) Running mc tests ... done in 7 minutes and 34 seconds
(7/13) Running minio-dotnet tests ... FAILED in 5 minutes and 5 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.AreEqual failed. Expected:<0x8D619813780B094-1>. Actual:<0x8D6198136FCBADB-1>. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.AreEqual failed. Expected:<0x8D619813780B094-1>. Actual:<0x8D6198136FCBADB-1>. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual[T](T expected, T actual, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.CopyObject_Test3(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 960
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 223

Executed 6 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#7ecba92e-be83-4fb8-85cc-9b3a3141a71d) |
<a name="7ecba92e-be83-4fb8-85cc-9b3a3141a71d"></a>
### 6406-e8c75cd/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:31631
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp aa9c71fe093c:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 2 minutes and 56 seconds
(2/13) Running aws-sdk-go tests ... done in 1 seconds
(3/13) Running aws-sdk-java tests ... done in 5 seconds
(4/13) Running aws-sdk-php tests ... done in 47 seconds
(5/13) Running aws-sdk-ruby tests ... done in 37 seconds
(6/13) Running mc tests ... done in 7 minutes and 42 seconds
(7/13) Running minio-dotnet tests ... FAILED in 5 minutes and 5 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.AreEqual failed. Expected:<0x8D619FF17E16878-1>. Actual:<0x8D619FF175AB9C1-1>. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.AreEqual failed. Expected:<0x8D619FF17E16878-1>. Actual:<0x8D619FF175AB9C1-1>. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual[T](T expected, T actual, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.CopyObject_Test3(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 960
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 223

Executed 6 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#8ed2a5fa-1b59-482f-a68e-ccb7387889d2) |
<a name="8ed2a5fa-1b59-482f-a68e-ccb7387889d2"></a>
### 6406-748a598/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:32071
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e303e316ec07:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 2 minutes and 46 seconds
(2/13) Running aws-sdk-go tests ... done in 2 seconds
(3/13) Running aws-sdk-java tests ... done in 1 seconds
(4/13) Running aws-sdk-php tests ... done in 43 seconds
(5/13) Running aws-sdk-ruby tests ... done in 42 seconds
(6/13) Running mc tests ... done in 8 minutes and 2 seconds
(7/13) Running minio-dotnet tests ... FAILED in 5 minutes and 12 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.AreEqual failed. Expected:<0x8D61A1466388206-1>. Actual:<0x8D61A1465C537E5-1>. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.AreEqual failed. Expected:<0x8D61A1466388206-1>. Actual:<0x8D61A1465C537E5-1>. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual[T](T expected, T actual, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.CopyObject_Test3(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 960
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 223

Executed 6 out of 13 tests successfully.
```

**Comment by nitisht** [comment]
Can this be closed?

**Comment by harshavardhana** [comment]
> Can this be closed?

which one goes first? @sinhaashish i see two PRs

**Comment by sinhaashish** [comment]
@harshavardhana @nitisht  This PR is closed . PR #6491  is the one which is has JSON select as well as refactored code.

**Pull request closed without merging**